frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/common-template-validations/selectors.ts

```diff
@@ -0,0 +1,114 @@
+import { iGetIn } from '../../../../../utils/immutable';
+import {
+  TEMPLATE_TYPE_LABEL,
+  TEMPLATE_WORKLOAD_LABEL,
+  TEMPLATE_OS_LABEL,
+  TEMPLATE_FLAVOR_LABEL,
+} from '../../../../../constants/vm';
+import {
+  CommonTemplatesValidation,
+  IntegerValidtionType,
+  CommonTemplatesValidationAttr,
+  IntegerValidationObj,
+  IntegerBetween,
+} from './validation-types';
+
+export const getRelevantTemplates = (commonTemplates, os, workloadProfile, flavor) => {
+  if (!commonTemplates || commonTemplates.length === 0 || !os || !workloadProfile) {
+    return [];
+  }
+  return (commonTemplates || []).filter(
+    (template) =>
+      iGetIn(template, ['metadata', 'labels', TEMPLATE_TYPE_LABEL]) === 'base' &&
+      (!os || iGetIn(template, ['metadata', 'labels', `${TEMPLATE_OS_LABEL}/${os}`])) &&
+      (!workloadProfile ||
+        iGetIn(template, [
+          'metadata',
+          'labels',
+          `${TEMPLATE_WORKLOAD_LABEL}/${workloadProfile}`,
+        ])) &&
+      (flavor === 'Custom' ||
+        iGetIn(template, ['metadata', 'labels', `${TEMPLATE_FLAVOR_LABEL}/${flavor}`])),
+  );
+};
+
+export const getValidationsFromTemplates = (templates): CommonTemplatesValidation[][] => {
+  return templates.map((relevantTemplate) =>
+    JSON.parse(iGetIn(relevantTemplate, ['metadata', 'annotations', 'validations'])),
+  );
+};
+
+export const getFieldValidations = (
+  validations: CommonTemplatesValidation[][],
+  jsonPath: string,
+): CommonTemplatesValidation[][] => {
+  return validations.map((validationArray: CommonTemplatesValidation[]) => {
+    return validationArray.filter((validation) => {
+      return validation.path.includes(jsonPath);
+    });
+  });
+};
+
+export const getValidationAttributes = (validation: CommonTemplatesValidation): string[] => {
+  const validationAttrs: string[] = [];
+  for (const attr in CommonTemplatesValidationAttr) {
+    if (attr in validation) {
+      validationAttrs.push(attr);
+    }
+  }
+  return validationAttrs;
+};
+
+export const getMinIntegerValidationObj = (
+  value: number,
+  message: string,
+): IntegerValidationObj => {
+  return { type: IntegerValidtionType.min, value, message };
+};
+
+export const getMaxIntegerValidationObj = (
+  value: number,
+  message: string,
+): IntegerValidationObj => {
+  return { type: IntegerValidtionType.max, value, message };
+};
+
+export const getBetweenIntegerValidationObj = (
+  value: IntegerBetween,
+  message: string,
+): IntegerValidationObj => {
+  return { type: IntegerValidtionType.between, value, message };
+};
+
+export const validateMinRule = (ivo: IntegerValidationObj, value: number): boolean => {
```

Comment:
I don't think we need all of these wrappers.
I think we could rather do one loop and a set two variables according to the min/max/between

```js
let min = 0;
let max = Number.POSITIVE_INFINITY;

loop over attributes and set accordingly
```

Reply:
hey, my 2 cents are that we do not need this set of methods at all, instead of calling them, we shoud do:
```
rule.min >= value 
```
all this checks are already valid after we check `getType(rule)` becase if type is `minType` rule must have a `rule.min` attribute ...